File: benchmarks/oop_benchmark/oop.py
class Circle:
    def __init__(self, radius):
        self.radius = radius
    def area(self):
        return 314 * self.radius * self.radius // 100
    def perimeter(self):
        return 628 * self.radius // 100

class Rectangle:
    def __init__(self, width, height):
        self.width = width
        self.height = height
    def area(self):
        return self.width * self.height
    def perimeter(self):
        return 2 * (self.width + self.height)

N = 100000
total_area = 0
total_perimeter = 0

for i in range(N):
    r = 1 + i % 100

    if i % 3 == 0:
        c = Circle(r)
        total_area += c.area()
        total_perimeter += c.perimeter()
    elif i % 3 == 1:
        rect = Rectangle(r, r * 2)
        total_area += rect.area()
        total_perimeter += rect.perimeter()
    else:
        sq = Rectangle(r, r)
        total_area += sq.area()
        total_perimeter += sq.perimeter()

print(total_area)
print(total_perimeter)

File: benchmarks/oop_benchmark/Oop.java
public class Oop {
    static class Circle {
        int radius;
        Circle(int r) { this.radius = r; }
        int area() { return 314 * radius * radius / 100; }
        int perimeter() { return 628 * radius / 100; }
    }

    static class Rectangle {
        int width, height;
        Rectangle(int w, int h) { this.width = w; this.height = h; }
        int area() { return width * height; }
        int perimeter() { return 2 * (width + height); }
    }

    public static void main(String[] args) {
        int N = 100000;
        long totalArea = 0;
        long totalPerimeter = 0;

        for (int i = 0; i < N; i++) {
            int r = 1 + i % 100;

            if (i % 3 == 0) {
                Circle c = new Circle(r);
                totalArea += c.area();
                totalPerimeter += c.perimeter();
            } else if (i % 3 == 1) {
                Rectangle rect = new Rectangle(r, r * 2);
                totalArea += rect.area();
                totalPerimeter += rect.perimeter();
            } else {
                Rectangle sq = new Rectangle(r, r);
                totalArea += sq.area();
                totalPerimeter += sq.perimeter();
            }
        }

        System.out.println(totalArea);
        System.out.println(totalPerimeter);
    }
}
